modules/k6/src/main/java/org/testcontainers/k6/K6Container.java

```diff
@@ -0,0 +1,98 @@
+package org.testcontainers.k6;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class K6Container extends GenericContainer<K6Container> {
+
+    /** Standard image for k6, as provided by Grafana. */
+    public static final DockerImageName K6_IMAGE = DockerImageName.parse("grafana/k6:0.49.0");
+
+    /**
+     * Extended image allowing for dynamic inclusion of k6 Extensions.
+     * @see <a href="https://grafana.com/docs/k6/latest/extensions/">k6 Extensions</a>
+     */
+    public static final DockerImageName K6_BUILDER_IMAGE = DockerImageName.parse("szkiba/k6x:v0.4.0");
+
+    private String testScript;
+
+    private List<String> cmdOptions = new ArrayList<>();
+
+    private Map<String, String> scriptVars = new HashMap<>();
+
+    /**
+     * Creates a new container instance based upon the {@link #K6_IMAGE}.
+     */
+    public K6Container() {
+        this(K6_IMAGE);
+    }
+
+    /**
+     * Creates a new container instance based upon the provided image.
+     */
+    public K6Container(DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(K6_IMAGE, K6_BUILDER_IMAGE);
+
+        setWaitStrategy(Wait.defaultWaitStrategy());
+    }
+
+    /**
+     * Specifies the test script to be executed within the container.
+     * @param scriptPath location of script to be copied into the container
+     * @return the builder
+     */
+    public K6Container withTestScript(String scriptPath) {
+        this.testScript = "/home/k6/" + scriptPath;
+        final MountableFile mountableFile = MountableFile.forClasspathResource(scriptPath);
+        withCopyFileToContainer(mountableFile, this.testScript);
+        return self();
+    }
```

Comment:
I think it is too restrict to use `MountableFile.forClasspathResource` and the Testcontainers API is flexible enough https://java.testcontainers.org/features/files/#copying-files to do so. IMO having a test demonstrating this would be enough but not offering an specific method to load files just from the classpath